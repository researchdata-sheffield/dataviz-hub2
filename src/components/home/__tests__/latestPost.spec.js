/**
 * @jest-environment jsdom
 */
/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import React from "react";
import { render, queryByText } from "@testing-library/react";
import LatestPost from "../latestPost";

jest.mock("../../../utils/shared", () => {
  return {
    ...jest.requireActual("../../../utils/shared"),
    getImageSource: () => {
      return null;
    }
  };
});

const mockData = {
  edges: [
    {
      node: {
        id: "4ed7c949-552f-594d-bb02-3b47dc104629",
        frontmatter: {
          type: null,
          title: "Mock featured Item 1",
          date: "15 Jan 2021",
          description: "Mock Description",
          tag: ["Mock tag"],
          category: ["Articles", "Events"],
          featured: true,
          thumbnail: null,
          author: [
            {
              name: "MOck Author",
              avatar: {
                childImageSharp: {
                  gatsbyImageData: {
                    layout: "fullWidth",
                    images: {
                      sources: [
                        {
                          srcSet:
                            "/static/0b4fd531c4d7e2b13cad834b37cad1fc/dacf8/Y.L_Weng.webp 591w",
                          type: "image/webp",
                          sizes: "100vw"
                        }
                      ]
                    },
                    width: 1,
                    height: 0.9475465313028765
                  }
                }
              }
            }
          ]
        },
        fields: {
          slug: "/blog/15/06/2021/Visualising-Cultural-Heritage-Data",
          readingTime: {
            text: "8 min read"
          }
        }
      }
    },
    {
      node: {
        id: "50198fed-ab7b-54d1-94b8-4f6529debeed",
        frontmatter: {
          type: null,
          title: "Mock featured Item 2",
          date: "16 March 2021",
          description:
            "The title says it all! Some snippets of advice for making our charts more fit for purpose.",
          tag: ["Tutorial"],
          category: ["Articles"],
          featured: true,
          thumbnail: null,
          author: [
            {
              name: "Gemma Ives",
              avatar: {
                childImageSharp: {
                  gatsbyImageData: {
                    layout: "fullWidth",
                    images: {
                      sources: [
                        {
                          srcSet:
                            "/static/c8d0f44f49a0c2ce38d557f5b9070c49/bf733/PPCirc.webp 626w",
                          type: "image/webp",
                          sizes: "100vw"
                        }
                      ]
                    },
                    width: 1,
                    height: 1
                  }
                }
              }
            }
          ]
        },
        fields: {
          slug: "/blog/16/03/2021/Good-Practice-Guide/Index",
          readingTime: {
            text: "11 min read"
          }
        }
      }
    },
    {
      node: {
        id: "9693afb1-c1fa-5871-98f2-f617ff11f2d7",
        frontmatter: {
          type: null,
          title: "Useful Resources for R",
          date: "18 February 2021",
          description:
            "Part 4 | A summary of resources that you might find useful when using R.",
          tag: ["R"],
          category: ["Articles"],
          featured: true,
          thumbnail: null,
          author: [
            {
              name: "Yu Liang Weng",
              avatar: {
                childImageSharp: {
                  gatsbyImageData: {
                    layout: "fullWidth",
                    images: {
                      sources: [
                        {
                          srcSet:
                            "/static/0b4fd531c4d7e2b13cad834b37cad1fc/dacf8/Y.L_Weng.webp 591w",
                          type: "image/webp",
                          sizes: "100vw"
                        }
                      ]
                    },
                    width: 1,
                    height: 0.9475465313028765
                  }
                }
              }
            }
          ]
        },
        fields: {
          slug: "/blog/18/02/2021/Useful-Resources-for-R",
          readingTime: {
            text: "6 min read"
          }
        }
      }
    },
    {
      node: {
        id: "11028bda-de4f-5638-b10b-3867d82f51d4",
        frontmatter: {
          type: null,
          title: "Templates for Shiny applications",
          date: "05 February 2021",
          description:
            "A simple framework to get you started creating your own Shiny app. Ready to use Shiny templates where you can replace elements with your own data and visualisations.",
          tag: [
            "Shiny",
            "Template",
            "R",
            "Mock tag 2",
            "Mock tag 3",
            "Mock tag 4"
          ],
          category: ["Tutorial"],
          featured: null,
          thumbnail: {
            childImageSharp: {
              gatsbyImageData: {
                layout: "fullWidth",
                images: {
                  sources: [
                    {
                      srcSet:
                        "/static/ab155ee272c11bb6b70e17943518efd4/56234/Thumb.webp 750w,\n/static/ab155ee272c11bb6b70e17943518efd4/b36af/Thumb.webp 1080w,\n/static/ab155ee272c11bb6b70e17943518efd4/918e5/Thumb.webp 1310w",
                      type: "image/webp",
                      sizes: "100vw"
                    }
                  ]
                },
                width: 1,
                height: 0.7923664122137405
              }
            }
          },
          author: [
            {
              name: "Gemma Ives",
              avatar: {
                childImageSharp: {
                  gatsbyImageData: {
                    layout: "fullWidth",
                    images: {
                      sources: [
                        {
                          srcSet:
                            "/static/c8d0f44f49a0c2ce38d557f5b9070c49/bf733/PPCirc.webp 626w",
                          type: "image/webp",
                          sizes: "100vw"
                        }
                      ]
                    },
                    width: 1,
                    height: 1
                  }
                }
              }
            },
            {
              name: "Yu Liang Weng",
              avatar: {
                childImageSharp: {
                  gatsbyImageData: {
                    layout: "fullWidth",
                    images: {
                      sources: [
                        {
                          srcSet:
                            "/static/0b4fd531c4d7e2b13cad834b37cad1fc/dacf8/Y.L_Weng.webp 591w",
                          type: "image/webp",
                          sizes: "100vw"
                        }
                      ]
                    },
                    width: 1,
                    height: 0.9475465313028765
                  }
                }
              }
            }
          ]
        },
        fields: {
          slug: "/blog/05/02/2021/Shiny-Template",
          readingTime: {
            text: "9 min read"
          }
        }
      }
    }
  ]
};

describe("Home | Latest Post", () => {
  let latest;

  beforeAll(() => {
    jest.spyOn(console, "warn").mockImplementation(() => {});
    jest.spyOn(console, "error").mockImplementation(() => {});
  });

  it("should renders correctly", async () => {
    latest = render(<LatestPost post={mockData} />);
    expect(latest).toMatchSnapshot();
  });

  it("should skip first two featured items", async () => {
    latest = render(<LatestPost post={mockData} />);

    const item1 = await queryByText(latest.container, "Mock feature Item 1");
    expect(item1).not.toBeInTheDocument();

    const item2 = await queryByText(latest.container, "Mock feature Item 2");
    expect(item2).not.toBeInTheDocument();
  });
});
