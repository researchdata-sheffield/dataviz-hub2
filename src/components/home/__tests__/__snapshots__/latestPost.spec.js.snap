// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home | Latest Post should renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full flex flex-wrap"
      >
        <div
          class="group w-full relative z-10 border-t-1 border-transparent first:border-gray-900"
          tabindex="0"
        >
          <div
            class="react-reveal w-full bg-gray-900 hover:bg-transparent text-white 2xl:text-xl"
          >
            <a
              class="flex flex-wrap flex-col md:flex-row justify-between w-full text-gray-500 hover:text-white px-5 lg:px-12 py-4"
              gatsby="true"
              href="/blog/18/02/2021/Useful-Resources-for-R"
            >
              <div
                class="flex flex-wrap items-center md:w-2/5 lg:w-3/12"
              >
                <svg
                  class="text-red-700 lg:text-white group-hover:text-red-700 text-3xl mr-1"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M20 4H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V6c0-1.11-.89-2-2-2zM8.5 15H7.3l-2.55-3.5V15H3.5V9h1.25l2.5 3.5V9H8.5v6zm5-4.74H11v1.12h2.5v1.26H11v1.11h2.5V15h-4V9h4v1.26zm7 3.74c0 .55-.45 1-1 1h-4c-.55 0-1-.45-1-1V9h1.25v4.51h1.13V9.99h1.25v3.51h1.12V9h1.25v5z"
                  />
                </svg>
                <a
                  class="sc-jRQBWg ertoVv rounded-full py-0 my-0 mx-1 border-none bg-white text-black hover:bg-gray-200 font-semibold invisible group-hover:visible"
                  gatsby="true"
                  href="/blog/category/articles"
                >
                  Articles
                </a>
              </div>
              <div
                class="md:w-3/5 lg:w-9/12 flex justify-between"
              >
                <div
                  class="inline-block font-semibold"
                >
                  Useful Resources for R
                </div>
                <div
                  class="inline-block font-semibold"
                >
                  18 Feb
                </div>
              </div>
            </a>
          </div>
          <div
            class="bg-white hidden xl:block fixed left-0 top-0 opacity-0 group-focus:opacity-100 group-hover:opacity-100 transform -translate-x-110% group-hover:translate-x-0 transition duration-500 shadow-xl"
            style="width: 33.333333%;"
          >
            <div
              class="min-h-50 max-h-50 w-full overflow-hidden"
            />
            <div
              class="flex flex-col bg-white min-h-50 py-6 px-8"
            >
              <p
                class="text-sm text-gray-500 font-normal 2xl:text-lg"
              >
                Yu Liang Weng
                 
                  |   
                6 min read
              </p>
              <p
                class="mt-2 text-gray-500"
              >
                <a
                  class="sc-gKclnd fLBWlL text-xs py-0 mr-1 my-0"
                  gatsby="true"
                  href="/blog/tag/r"
                >
                  R
                </a>
              </p>
              <p
                class="mt-4 2xl:text-2xl text-black"
              >
                Part 4 | A summary of resources that you might find useful when using R.
              </p>
            </div>
          </div>
        </div>
        <div
          class="group w-full relative z-10 border-t-1 border-transparent first:border-gray-900"
          tabindex="0"
        >
          <div
            class="react-reveal w-full bg-gray-900 hover:bg-transparent text-white 2xl:text-xl"
          >
            <a
              class="flex flex-wrap flex-col md:flex-row justify-between w-full text-gray-500 hover:text-white px-5 lg:px-12 py-4"
              gatsby="true"
              href="/blog/05/02/2021/Shiny-Template"
            >
              <div
                class="flex flex-wrap items-center md:w-2/5 lg:w-3/12"
              >
                <svg
                  class="text-red-700 lg:text-white group-hover:text-red-700 text-3xl mr-1"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M20 4H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V6c0-1.11-.89-2-2-2zM8.5 15H7.3l-2.55-3.5V15H3.5V9h1.25l2.5 3.5V9H8.5v6zm5-4.74H11v1.12h2.5v1.26H11v1.11h2.5V15h-4V9h4v1.26zm7 3.74c0 .55-.45 1-1 1h-4c-.55 0-1-.45-1-1V9h1.25v4.51h1.13V9.99h1.25v3.51h1.12V9h1.25v5z"
                  />
                </svg>
                <a
                  class="sc-jRQBWg ertoVv rounded-full py-0 my-0 mx-1 border-none bg-white text-black hover:bg-gray-200 font-semibold invisible group-hover:visible"
                  gatsby="true"
                  href="/blog/category/tutorial"
                >
                  Tutorial
                </a>
              </div>
              <div
                class="md:w-3/5 lg:w-9/12 flex justify-between"
              >
                <div
                  class="inline-block font-semibold"
                >
                  Templates for Shiny applications
                </div>
                <div
                  class="inline-block font-semibold"
                >
                  05 Feb
                </div>
              </div>
            </a>
          </div>
          <div
            class="bg-white hidden xl:block fixed left-0 top-0 opacity-0 group-focus:opacity-100 group-hover:opacity-100 transform -translate-x-110% group-hover:translate-x-0 transition duration-500 shadow-xl"
            style="width: 33.333333%;"
          >
            <div
              class="min-h-50 max-h-50 w-full overflow-hidden"
            />
            <div
              class="flex flex-col bg-white min-h-50 py-6 px-8"
            >
              <p
                class="text-sm text-gray-500 font-normal 2xl:text-lg"
              >
                Gemma Ives · 
                Yu Liang Weng
                 
                  |   
                9 min read
              </p>
              <p
                class="mt-2 text-gray-500"
              >
                <a
                  class="sc-gKclnd fLBWlL text-xs py-0 mr-1 my-0"
                  gatsby="true"
                  href="/blog/tag/shiny"
                >
                  Shiny
                </a>
                <a
                  class="sc-gKclnd fLBWlL text-xs py-0 mr-1 my-0"
                  gatsby="true"
                  href="/blog/tag/template"
                >
                  Template
                </a>
                <a
                  class="sc-gKclnd fLBWlL text-xs py-0 mr-1 my-0"
                  gatsby="true"
                  href="/blog/tag/r"
                >
                  R
                </a>
                <a
                  class="sc-gKclnd fLBWlL text-xs py-0 mx-1 my-0"
                  gatsby="true"
                  href="/blog/05/02/2021/Shiny-Template"
                >
                  +
                  3
                   more
                </a>
              </p>
              <p
                class="mt-4 2xl:text-2xl text-black"
              >
                A simple framework to get you started creating your own Shiny app. Ready to use Shiny templates where you can replace elements with your own data and visualisations.
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex flex-wrap w-full bg-gray-900 py-10 px-3 justify-center items-center self-center z-10"
        style="transition: .3s ease;"
      >
        <div
          class="text-white mr-10"
        >
          Discover more articles and tutorials at our blog.
        </div>
        <a
          gatsby="true"
          href="/blog"
        >
          <button
            class="sc-kDTinF erPJFv group py-2 px-4 mt-6 md:mt-0 text-sm bg-white text-black hover:text-white"
            title="read more on blog"
            type="AnimateButton"
          >
            Read more
            <svg
              class="inline-block transform group-hover:translate-x-1 transition duration-100"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
              />
            </svg>
          </button>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full flex flex-wrap"
    >
      <div
        class="group w-full relative z-10 border-t-1 border-transparent first:border-gray-900"
        tabindex="0"
      >
        <div
          class="react-reveal w-full bg-gray-900 hover:bg-transparent text-white 2xl:text-xl"
        >
          <a
            class="flex flex-wrap flex-col md:flex-row justify-between w-full text-gray-500 hover:text-white px-5 lg:px-12 py-4"
            gatsby="true"
            href="/blog/18/02/2021/Useful-Resources-for-R"
          >
            <div
              class="flex flex-wrap items-center md:w-2/5 lg:w-3/12"
            >
              <svg
                class="text-red-700 lg:text-white group-hover:text-red-700 text-3xl mr-1"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M20 4H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V6c0-1.11-.89-2-2-2zM8.5 15H7.3l-2.55-3.5V15H3.5V9h1.25l2.5 3.5V9H8.5v6zm5-4.74H11v1.12h2.5v1.26H11v1.11h2.5V15h-4V9h4v1.26zm7 3.74c0 .55-.45 1-1 1h-4c-.55 0-1-.45-1-1V9h1.25v4.51h1.13V9.99h1.25v3.51h1.12V9h1.25v5z"
                />
              </svg>
              <a
                class="sc-jRQBWg ertoVv rounded-full py-0 my-0 mx-1 border-none bg-white text-black hover:bg-gray-200 font-semibold invisible group-hover:visible"
                gatsby="true"
                href="/blog/category/articles"
              >
                Articles
              </a>
            </div>
            <div
              class="md:w-3/5 lg:w-9/12 flex justify-between"
            >
              <div
                class="inline-block font-semibold"
              >
                Useful Resources for R
              </div>
              <div
                class="inline-block font-semibold"
              >
                18 Feb
              </div>
            </div>
          </a>
        </div>
        <div
          class="bg-white hidden xl:block fixed left-0 top-0 opacity-0 group-focus:opacity-100 group-hover:opacity-100 transform -translate-x-110% group-hover:translate-x-0 transition duration-500 shadow-xl"
          style="width: 33.333333%;"
        >
          <div
            class="min-h-50 max-h-50 w-full overflow-hidden"
          />
          <div
            class="flex flex-col bg-white min-h-50 py-6 px-8"
          >
            <p
              class="text-sm text-gray-500 font-normal 2xl:text-lg"
            >
              Yu Liang Weng
               
                |   
              6 min read
            </p>
            <p
              class="mt-2 text-gray-500"
            >
              <a
                class="sc-gKclnd fLBWlL text-xs py-0 mr-1 my-0"
                gatsby="true"
                href="/blog/tag/r"
              >
                R
              </a>
            </p>
            <p
              class="mt-4 2xl:text-2xl text-black"
            >
              Part 4 | A summary of resources that you might find useful when using R.
            </p>
          </div>
        </div>
      </div>
      <div
        class="group w-full relative z-10 border-t-1 border-transparent first:border-gray-900"
        tabindex="0"
      >
        <div
          class="react-reveal w-full bg-gray-900 hover:bg-transparent text-white 2xl:text-xl"
        >
          <a
            class="flex flex-wrap flex-col md:flex-row justify-between w-full text-gray-500 hover:text-white px-5 lg:px-12 py-4"
            gatsby="true"
            href="/blog/05/02/2021/Shiny-Template"
          >
            <div
              class="flex flex-wrap items-center md:w-2/5 lg:w-3/12"
            >
              <svg
                class="text-red-700 lg:text-white group-hover:text-red-700 text-3xl mr-1"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M20 4H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V6c0-1.11-.89-2-2-2zM8.5 15H7.3l-2.55-3.5V15H3.5V9h1.25l2.5 3.5V9H8.5v6zm5-4.74H11v1.12h2.5v1.26H11v1.11h2.5V15h-4V9h4v1.26zm7 3.74c0 .55-.45 1-1 1h-4c-.55 0-1-.45-1-1V9h1.25v4.51h1.13V9.99h1.25v3.51h1.12V9h1.25v5z"
                />
              </svg>
              <a
                class="sc-jRQBWg ertoVv rounded-full py-0 my-0 mx-1 border-none bg-white text-black hover:bg-gray-200 font-semibold invisible group-hover:visible"
                gatsby="true"
                href="/blog/category/tutorial"
              >
                Tutorial
              </a>
            </div>
            <div
              class="md:w-3/5 lg:w-9/12 flex justify-between"
            >
              <div
                class="inline-block font-semibold"
              >
                Templates for Shiny applications
              </div>
              <div
                class="inline-block font-semibold"
              >
                05 Feb
              </div>
            </div>
          </a>
        </div>
        <div
          class="bg-white hidden xl:block fixed left-0 top-0 opacity-0 group-focus:opacity-100 group-hover:opacity-100 transform -translate-x-110% group-hover:translate-x-0 transition duration-500 shadow-xl"
          style="width: 33.333333%;"
        >
          <div
            class="min-h-50 max-h-50 w-full overflow-hidden"
          />
          <div
            class="flex flex-col bg-white min-h-50 py-6 px-8"
          >
            <p
              class="text-sm text-gray-500 font-normal 2xl:text-lg"
            >
              Gemma Ives · 
              Yu Liang Weng
               
                |   
              9 min read
            </p>
            <p
              class="mt-2 text-gray-500"
            >
              <a
                class="sc-gKclnd fLBWlL text-xs py-0 mr-1 my-0"
                gatsby="true"
                href="/blog/tag/shiny"
              >
                Shiny
              </a>
              <a
                class="sc-gKclnd fLBWlL text-xs py-0 mr-1 my-0"
                gatsby="true"
                href="/blog/tag/template"
              >
                Template
              </a>
              <a
                class="sc-gKclnd fLBWlL text-xs py-0 mr-1 my-0"
                gatsby="true"
                href="/blog/tag/r"
              >
                R
              </a>
              <a
                class="sc-gKclnd fLBWlL text-xs py-0 mx-1 my-0"
                gatsby="true"
                href="/blog/05/02/2021/Shiny-Template"
              >
                +
                3
                 more
              </a>
            </p>
            <p
              class="mt-4 2xl:text-2xl text-black"
            >
              A simple framework to get you started creating your own Shiny app. Ready to use Shiny templates where you can replace elements with your own data and visualisations.
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-wrap w-full bg-gray-900 py-10 px-3 justify-center items-center self-center z-10"
      style="transition: .3s ease;"
    >
      <div
        class="text-white mr-10"
      >
        Discover more articles and tutorials at our blog.
      </div>
      <a
        gatsby="true"
        href="/blog"
      >
        <button
          class="sc-kDTinF erPJFv group py-2 px-4 mt-6 md:mt-0 text-sm bg-white text-black hover:text-white"
          title="read more on blog"
          type="AnimateButton"
        >
          Read more
          <svg
            class="inline-block transform group-hover:translate-x-1 transition duration-100"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
            />
          </svg>
        </button>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
